esi_leap/api/controllers/v1/contract.py

```diff
@@ -54,9 +54,12 @@ class ContractsController(rest.RestController):
     def get_one(self, contract_uuid):
         request = pecan.request.context
         cdict = request.to_policy_values()
-        policy.authorize('esi_leap:contract:get', cdict, cdict)
+        policy.authorize('esi_leap:contract:list', cdict, cdict)
 
-        c = contract.Contract.get(request, contract_uuid)
+        filters = ContractsController._contract_get_one_filters(
+            cdict, request.project_id)
```

Comment:
Are we really just checking ownership? If so, I think we should call out the check more explicitly (and in the delete function), by which I mean: retrieve the contract, and then call a function that checks permissions.

We also don't really want to apply a filter to a 'get_one' call; filtering really only applies to lists.

I also don't think this check needs to be a separate function; after getting the contract, you just need:

```
if not verify_admin and c.project_id != request.project_id:
    throw WhateverException
```

Reply:
I wanted to remove all the checks for being an admin and protecting data to outside of the dbapi

Reply:
Definitely - I'm not saying you need to put that check into the database. But you can retrieve the contract, and *then* check permissions in the REST API.

Reply:
That wouldnt be possible for the contract get. Because the contract returned will not say anything about the owner of the related offer

Reply:
Unless we did a second call to the dbapi to then get the related offer. Doing it this way is just one call

Reply:
Having two calls is fine. If you apply a filter, you're essentially still having the db enforce permissions. If we're just going to enforce permissions in one place, it should be here at the REST API.

Reply:
well the db doesnt have to be fed a filter to work. The db doesn't enforce any permissions, the api just enforces how the db is called

Reply:
If by enforce, you mean do the dirty work of surreptitiously producing no data then I suppose the db is enforcing

Reply:
Yep, that's exactly what I mean - if for whatever reason we decide to loosen restrictions later, it'd be useful to have all the controlling logic be at the REST API level.